src/components/ui4/ButtonUi4.tsx

```diff
@@ -141,16 +146,18 @@ export function ButtonUi4(props: Props) {
   const finalContainerCommon = [styles.containerCommon, containerStyle, customMargin, customPadding]
 
   return (
-    <TouchableOpacity disabled={disabled || pending} style={finalContainerCommon} onPress={handlePress} testID={testID}>
-      <LinearGradient
-        {...gradientProps}
-        style={[styles.contentCommon, dynamicGradientStyles, mini ? styles.contentSizeMini : styles.contentSizeDefault, ...finalContainerCommon]}
-      >
-        {hideContent ? null : children}
-        {hideContent ? null : maybeText}
-        {!hideContent ? null : <ActivityIndicator color={spinnerColor} style={styles.spinnerCommon} />}
-      </LinearGradient>
-    </TouchableOpacity>
+    <MaybeEdgeAnim when={animProps != null} {...animProps}>
```

Comment:
This layering makes no sense. The button user should just do `<EdgeAnim><ButtonUi4/></EdgeAnim>` themselves. There is no reason for the button to be animation-aware.

At the very least, the `ButtonsViewUi4` should be in charge of adding the `<EdgeAnim>` wrapper.

Reply:
We have this mix of requirements right now. 

In the ideal future, no scene touches `ButtonUi4` (or `MainButton` which is just a `ButtonUi4` wrapper) but we do now because of time restrictions. Since we have a mix of scenes where they sometimes use `ButtonUi4`/`MainButton` directly and other times `ButtonsViewUi4`, I had to add it to both.